src/services/utxoDiffSincePoint.ts

```diff
@@ -148,68 +104,56 @@ const buildSelectFromForOutputs = () => {
   INNER JOIN tx_out ON tx.id = tx_out.tx_id`;
 };
 
-const buildWhereClause = (validContract: boolean, useItemIndex: boolean) => {
+const buildWhereClause = (validContract: boolean) => {
   return `WHERE ${validContract ? "" : "NOT "}tx.valid_contract
   AND block.block_no <= ($3)::word31type
-  ${
-    useItemIndex
-      ? `AND (
-      block.block_no > ($4)::word31type
-      OR (
-        block.block_no >= ($4)::word31type
-        AND encode(tx.hash,'hex') = ($5)::varchar
-        AND tx_out.index > ($6)::txindex
-      )
-    )`
-      : "AND block.block_no > ($4)::word31type"
-  }
+  AND block.block_no > ($4)::word31type
   AND (
     tx_out.address = any(($1)::varchar array) 
     OR tx_out.payment_cred = any(($2)::bytea array)
   )`;
 };
 
-const buildInputQuery = (useItemIndex: boolean) => {
+const buildInputQuery = () => {
   return `${buildSelectColumns(DiffType.INPUT)}
   ${buildSelectFromForInputs()}
-  ${buildWhereClause(true, useItemIndex)}`;
+  ${buildWhereClause(true)}`;
 };
 
-const buildCollateralQuery = (useItemIndex: boolean) => {
+const buildCollateralQuery = () => {
   return `${buildSelectColumns(DiffType.INPUT)}
   ${buildSelectFromForCollaterals()}
-  ${buildWhereClause(false, useItemIndex)}`;
+  ${buildWhereClause(false)}`;
 };
 
-const buildOutputQuery = (useItemIndex: boolean) => {
+const buildOutputQuery = () => {
   return `${buildSelectColumns(DiffType.OUTPUT)}
   ${buildSelectFromForOutputs()}
-  ${buildWhereClause(true, useItemIndex)}`;
+  ${buildWhereClause(true)}`;
 };
 
-const buildFullQuery = (useItemIndex: boolean) => {
+const buildFullQuery = () => {
   return `SELECT * FROM (
-    ${buildInputQuery(useItemIndex)}
+    ${buildInputQuery()}
     UNION ALL
-    ${buildCollateralQuery(useItemIndex)}
+    ${buildCollateralQuery()}
     UNION ALL
-    ${buildOutputQuery(useItemIndex)}
+    ${buildOutputQuery()}
   ) as q
-  ORDER BY q."blockNumber", CASE q.type WHEN 'I' THEN 1 ELSE 0 END
-  LIMIT $${5 + (useItemIndex ? 2 : 0)}::word31type;`;
```

Comment:
@yushih  We cannot get rid of the limit completely, because the query is initially done for the range `[null, bestblock]` for example, which means it will try to return the entire history for the addresses. The idea is that it always must be **"not more than the $limit of items BUT ALSO only values before the $bestblock"**.

The change that you are proposing will not work exactly the way you are proposing it:

> My proposal in this PR is to circumvent the problem by paginating by the number of blocks instead of by the number of diff items. So instead of the diffLimit parameter, the query takes a blockCount parameter and only the specified number of blocks will be considered. Note this does mean that the number of returned diff items will vary, but since the number of transactions in a block and the number of inputs and outputs in a transaction are bound, the maximum number of returned diff items has an upper bound and the purpose of the pagination is also achieved.

The problem with the `blockCount` parameter is that the frontend then has to suggest to the backend how many blocks to search for any data, but the frontend has no idea how many blocks there are with data at all, and we want to avoid any pointless requests where, for example, frontend asks backend to get utxos from "next 1000 blocks only" but then there are no utxos for these addresses in the next 1000 blocks at all, and next 1000 and next as well, etc. The frontend will just endup sending requests over and over. And the entire history of the wallet then might turn up to be within a single 1000 blocks after that and then the backend will be forced to send a ton of data.

A potential solution for this would be to have a "soft limit" to the number of items that can be returned and then the backend would be considering full blocks but only until that soft limit number is reached or nearly reached. The backend then must be able to select **full blocks** until the number of itsems in the blocks is above the certain limit.